content/en/docs/develop/porting.md

```diff
@@ -614,23 +614,338 @@ To make a patch:
+
+In case you have pre-compiled object files, you could add them (but due to possible incompatible compilation flags of your pre-compiled object files, you should handle this with care):
+
+```Makefile
+APPNAME_OBJS-y += $(APPNAME_BASE)/path_to_src/myobj.o
+```
+
+You can also use `APPNAME_OBJS-y` to add pre-built libraries (as `.o` or `.a`):
+
+```Makefile
+APPNAME_OBJS-y += $(APPNAME_BASE)/path_to_lib/mylib.a
+```
+
+Once you have specified all of your source files (and optionally binary files), it is generally also necessary to specify include paths and compile flags:
+
+```Makefile
+# Include paths
+APPNAME_ASINCLUDES  += -I$(APPNAME_BASE)/path_to_include/include [for assembly files]
+APPNAME_CINCLUDES   += -I$(APPNAME_BASE)/path_to_include/include [for C files]
+APPNAME_CXXINCLUDES += -I$(APPNAME_BASE)/path_to_include/include [for C++ files]
+
+# Flags for application sources
+APPNAME_ASFLAGS-y   += -DFLAG_NAME1 ... -DFLAG_NAMEN [for assembly files]
+APPNAME_CFLAGS-y    += -DFLAG_NAME1 ... -DFLAG_NAMEN [for C files]
+APPNAME_CXXFLAGS-y  += -DFLAG_NAME1 ... -DFLAG_NAMEN [for C++ files]
+```
+
+With all of these in place, you can save `Makefile.uk`, and type `make`.
+Assuming that the chosen Unikraft libraries provide all of the support that your application needs, Unikraft should compile and link everything together, and output one image per target platform specified in the menu.
+
+In addition to all the functionality mentioned, applications might need to perform a number of additional tasks after the sources are downloaded and unpacked, but _before_ the compilation takes place (e.g., run a configure script or a custom script that generates source code from source files).
+To support this, Unikraft provides a "prepare" variable which you can set to a temporary marker file and from there to a target in your `Makefile.uk` file.
+For example:
+
+```Makefile
+$(APPNAME_BUILD)/.prepared: [dependencies to further targets]
+       cmd && $(TOUCH) $@
+
+UK_PREPARE += $(APPNAME_BUILD)/.prepared
+```
+
+In this way, you ensure that `cmd` is run before any compilation takes place.
+If you use `fetch`, add `$(APPNAME_BUILD)/.origin` as dependency.
+If you use `patch`, add `$(APPNAME_BUILD)/.patched` instead.
+
+Further, you may find it necessary to specify compile flags or includes only for a _specific_ source file.
+Unikraft supports this through the following syntax:
+
+```Makefile
+APPNAME_SRCS-y += $(APPNAME_BASE)/filename.c
+APPNAME_FILENAME_FLAGS-y += -DFLAG
+APPNAME_FILENAME_INCLUDES-y += -Iextra/include
+```
+
+It is also possible to compile a single source file multiple times with different flags.
+For this case, Unikraft supports variants:
+
+```Makefile
+APPNAME_SRCS-y += $(APPNAME_BASE)/filename.c|variantname
+APPNAME_FILENAME_VARIANTNAME_FLAGS-y += -DFLAG2
+APPNAME_FILENAME_VARIANTNAME_INCLUDES-y += -Iextra/include
+```
+
+**Note: The build system treats the reserved `isr` variant differently.
```

Comment:
Use the [Docsy `alert` shortcode](https://www.docsy.dev/docs/adding-content/shortcodes/#alert).